rpc/src/v1/helpers/dispatch.rs

```diff
@@ -69,12 +69,19 @@ pub trait Dispatcher: Send + Sync + Clone {
 	fn fill_optional_fields(&self, request: TransactionRequest, default_sender: Address, force_nonce: bool)
 		-> BoxFuture<FilledTransactionRequest>;
 
-	/// Sign the given transaction request without dispatching, fetching appropriate nonce.
-	fn sign(&self, accounts: Arc<AccountProvider>, filled: FilledTransactionRequest, password: SignWith)
-		-> BoxFuture<WithToken<SignedTransaction>>;
+	/// Sign the given transaction request, fetching appropriate nonce and executing the PostSign action
+	fn sign<P>(
+		&self,
+		accounts: Arc<AccountProvider>,
+		filled: FilledTransactionRequest,
+		password: SignWith,
+		post_sign: P
+	) -> BoxFuture<<P as PostSign>::Item>
```

Comment:
should not need typecast here
 
```suggestion
	) -> BoxFuture<P::Item>
```